src/self_update.sh

```diff
@@ -0,0 +1,209 @@
+include "${KW_LIB_DIR}/kwlib.sh"
+include "${KW_LIB_DIR}/kwio.sh"
+include "${KW_LIB_DIR}/help.sh"
+
+declare -gA options_values
+
+function self_update_main()
+{
+  local target_branch='master'
+  local path_to_tmp_dir
+  local cmd
+  local flag
+  local ret
+
+  parse_self_update_options "$@"
+  if [[ "$?" != 0 ]]; then
+    complain "Invalid option: ${options_values['ERROR']}"
+    self_update_help
+    return 22 # EINVAL
+  fi
+
+  if [[ -n "${options_values['UNSTABLE']}" ]]; then
+    target_branch='unstable'
+  fi
+
+  printf '%s\n' "Updating kw based on the branch ${target_branch}"
+  if [[ $(ask_yN 'Do you want to continue?') =~ '0' ]]; then
+    complain 'Update aborted.'
+    return 125 # ECANCELED
+  fi
+
+  flag="${options_values['TEST']}"
+  path_to_tmp_dir=$(mktemp --directory)
+  cmd="update_from_official_repo ${target_branch} ${path_to_tmp_dir}"
```

Comment:
`update_from_official_repo` is your function, you don't need to use `cmd_manager` for that. Usually, we use `cmd_manager` for a global system command like `cd`, `rm`, `find`, etc.

Reply:
The reason I used `cmd_manager` was to remove the logic (the `update_from_official_repo` call) from inside the `if` statement as you requested in the comment referenced below. Could not think of another way to check if it is in test mode (i.e. don't actually execute the function) without using an `if`.

> Fwiu, this `TEST` is only for unit test reasons; if so, it will be better if you invert the logic in a way that the code under this condition is moved outside this `if` statement. The reason for that is because `TEST` is not part of the normal code flow; it is more like an exception made only for test reasons.

Reply:
I did not get it... what is the problem with simply calling:

```
update_from_official_repo ${target_branch} ${path_to_tmp_dir}
if [[ "$?" != 0 ]]; then ...
```

Do you want to use `cmd_manager` for trying to test this main function? If you test `update_from_official_repo`, you don't need to test it via `self_update_main`.